SpatialGDK/Source/SpatialGDK/Public/Interop/Connection/ConnectionConfig.h

```diff
@@ -176,33 +174,52 @@ class FReceptionistConfig : public FConnectionConfig
 
 	bool TryLoadCommandLineArgs()
 	{
-		bool bSuccess = true;
 		const TCHAR* CommandLine = FCommandLine::Get();
 
+		// Get command line options first since the URL handling will modify the CommandLine string
+		FParse::Value(CommandLine, TEXT("receptionistPort"), ReceptionistPort);
+
 		// Parse the command line for receptionistHost, if it exists then use this as the host IP.
-		if (!FParse::Value(CommandLine, TEXT("receptionistHost"), ReceptionistHost))
+		FString Host;
+		if (!FParse::Value(CommandLine, TEXT("receptionistHost"), Host))
 		{
 			// If a receptionistHost is not specified then parse for an IP address as the first argument and use this instead.
 			// This is how native Unreal handles connecting to other IPs, a map name can also be specified, in this case we use the default IP.
 			FString URLAddress;
-			FParse::Token(CommandLine, URLAddress, 0);
-			FRegexPattern Ipv4RegexPattern(TEXT("^(?:[0-9]{1,3}\\.){3}[0-9]{1,3}$"));
-			FRegexMatcher IpV4RegexMatcher(Ipv4RegexPattern, *URLAddress);
-			bSuccess = IpV4RegexMatcher.FindNext();
-			if (bSuccess)
+			FParse::Token(CommandLine, URLAddress, false);
+			const FURL URL(nullptr, *URLAddress, TRAVEL_Absolute);
+			if (URL.Valid)
 			{
-				SetReceptionistHost(URLAddress);
+				SetReceptionistHost(URL);
 			}
 		}
+		else
+		{
+			SetReceptionistHost(Host);
+		}
 
-		FParse::Value(CommandLine, TEXT("receptionistPort"), ReceptionistPort);
-		return bSuccess;
+		return true;
+	}
+
+	void SetReceptionistHost(const FString& Host)
```

Comment:
I think we can hide this functionality by either make this function private or adding the functionality to the overloaded SetReceptionistHost.

Perhaps we should rename void SetReceptionistHost(const FURL& URL) to something like SetupFromURL.

Reply:
True, with all the additional changes this function can be made private now 👍